Common/src/element_linear.cpp

```diff
@@ -2609,118 +772,75 @@ CPRISM6::CPRISM6(unsigned short val_nDim, CConfig *config)
       val_Ni = 0.5*Zeta*(Xi+1.0);             GaussPoint[iGauss]->SetNi(val_Ni,4);
       val_Ni = 0.5*(1.0-Eta-Zeta)*(Xi+1.0);   GaussPoint[iGauss]->SetNi(val_Ni,5);
 
-  }
+      /*--- dN/d xi ---*/
 
-}
+      dNiXj[iGauss][0][0] = -0.5*Eta;
+      dNiXj[iGauss][1][0] = -0.5*Zeta;
+      dNiXj[iGauss][2][0] = -0.5*(1.0-Eta-Zeta);
+      dNiXj[iGauss][3][0] =  0.5*Eta;
+      dNiXj[iGauss][4][0] =  0.5*Zeta;
+      dNiXj[iGauss][5][0] =  0.5*(1.0-Eta-Zeta);
 
-CPRISM6::~CPRISM6(void) {
+      /*--- dN/d eta ---*/
 
-}
+      dNiXj[iGauss][0][1] =  0.5*(1.0-Xi);
+      dNiXj[iGauss][1][1] =  0.0;
+      dNiXj[iGauss][2][1] = -0.5*(1.0-Xi);
+      dNiXj[iGauss][3][1] =  0.5*(Xi+1.0);
+      dNiXj[iGauss][4][1] =  0.0;
+      dNiXj[iGauss][5][1] = -0.5*(Xi+1.0);
 
-void CPRISM6::ComputeGrad_Linear(void){
+      /*--- dN/d mu ---*/
 
-    su2double Xi, Eta, Zeta;
-    su2double Jacobian[3][3], dNiXj[8][3];
-    su2double detJac, GradNi_Xj;
-    su2double ad[3][3];
-    unsigned short iNode, iDim, jDim, iGauss;
+      dNiXj[iGauss][0][2] =  0.0;
+      dNiXj[iGauss][1][2] =  0.5*(1.0-Xi);
+      dNiXj[iGauss][2][2] = -0.5*(1.0-Xi);
+      dNiXj[iGauss][3][2] =  0.0;
+      dNiXj[iGauss][4][2] =  0.5*(Xi+1.0);
+      dNiXj[iGauss][5][2] = -0.5*(Xi+1.0);
 
-    for (iGauss = 0; iGauss < nGaussPoints; iGauss++){
+  }
 
-      Xi = GaussCoord[iGauss][0];
-      Eta = GaussCoord[iGauss][1];
-      Zeta = GaussCoord[iGauss][2];
+  /*--- Store the extrapolation functions ---*/
 
-      /*--- dN/d xi ---*/
+  su2double ExtrapCoord[6][3];
 
-      dNiXj[0][0] = -0.5*Eta;
-      dNiXj[1][0] = -0.5*Zeta;
-      dNiXj[2][0] = -0.5*(1.0-Eta-Zeta);
-      dNiXj[3][0] = 0.5*Eta;
-      dNiXj[4][0] = 0.5*Zeta;
-      dNiXj[5][0] = 0.5*(1.0-Eta-Zeta);
+  ExtrapCoord[0][0] = -1.732050807568877;  ExtrapCoord[0][1] = -1.0/3.0;  ExtrapCoord[0][2] = -1.0/3.0;
```

Comment:
As in other places in the code, a su2double Sqrt3 variable could be useful here.